olive/engine/footprint.py

```diff
@@ -58,8 +59,8 @@ def __init__(
         objective_dict: Dict = None,
         is_marked_pareto_frontier: bool = False,
     ):
-        self.nodes = nodes or OrderedDict()
-        self.objective_dict = objective_dict or {}
+        self.nodes = deepcopy(nodes) or OrderedDict()
+        self.objective_dict = deepcopy(objective_dict) or {}
```

Comment:
It would be better to move deepcopy into the create metthod? for other footprint creation, the deepcopy is needless if the node is temp created?